extensions/api/data-management/asset/src/main/java/org/eclipse/dataspaceconnector/api/datamanagement/asset/service/AssetQueryValidator.java

```diff
@@ -0,0 +1,48 @@
+/*
+ *  Copyright (c) 2022 Microsoft Corporation
+ *
+ *  This program and the accompanying materials are made available under the
+ *  terms of the Apache License, Version 2.0 which is available at
+ *  https://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  SPDX-License-Identifier: Apache-2.0
+ *
+ *  Contributors:
+ *       Microsoft Corporation - initial API and implementation
+ *
+ */
+
+package org.eclipse.dataspaceconnector.api.datamanagement.asset.service;
+
+import org.eclipse.dataspaceconnector.spi.query.QueryValidator;
+import org.eclipse.dataspaceconnector.spi.result.Result;
+import org.eclipse.dataspaceconnector.spi.types.domain.asset.Asset;
+
+import java.util.List;
+
+import static java.lang.String.format;
+
+class AssetQueryValidator extends QueryValidator {
+    private static final List<String> KNOWN_PROPERTIES = List.of(
+            Asset.PROPERTY_ID,
+            Asset.PROPERTY_NAME,
+            Asset.PROPERTY_DESCRIPTION,
+            Asset.PROPERTY_VERSION,
+            Asset.PROPERTY_CONTENT_TYPE
+    );
+
+    AssetQueryValidator(Class<Asset> assetClass) {
```

Comment:
This constructor could have no arguments and have the responsibility to pass the `Asset.class` object to the super constructor